widgets/src/main/java/org/pentaho/mantle/client/dialogs/scheduling/AdditionalDetailsPanel.java

```diff
@@ -0,0 +1,117 @@
+
+    private ListBox logLevelLb = createLogLevelListBox();
+    private ICallback<IChangeHandler> onChangeHandler = null;
+    private ErrorLabel detailsLabel = null;
+    private CheckBox enableSafeMode;
+    private CheckBox gatherMetrics;
+
+    private static final String SCHEDULER_ADDITIONAL_PANEL = "schedule-editor-additional-panel"; //$NON-NLS-1$
+    private static final String SCHEDULE_DETAILS_INPUT = "schedule-details-input";
+
+    public AdditionalDetailsPanel() {
+        CaptionPanel detailPanel = new CaptionPanel( Messages.getString( "schedule.detail" ) );
+        detailPanel.setStyleName( SCHEDULER_ADDITIONAL_PANEL );
+        detailPanel.getElement().setId( SCHEDULE_DETAILS_INPUT );
+
+        VerticalFlexPanel dVP = new VerticalFlexPanel();
+        enableSafeMode = new CheckBox();
+        enableSafeMode.setText( Messages.getString( "schedule.enableSafeMode" ) ); //$NON-NLS-1$
+        enableSafeMode.setStyleName( "enableSafeModeCheckbox" ); //$NON-NLS-1$
+        dVP.add( enableSafeMode );
+
+        gatherMetrics = new CheckBox();
+        gatherMetrics.setText( Messages.getString( "schedule.performanceMetrics" ) ); //$NON-NLS-1$
+        gatherMetrics.setStyleName( "gatherPerfmnceMetricsCheckbox" ); //$NON-NLS-1$
+        dVP.add( gatherMetrics );
+
+        HorizontalFlexPanel logLvlHP = new HorizontalFlexPanel();
+        logLevelLb.getElement().setId( "log-level-lb" );
+        logLvlHP.add( logLevelLb );
+        Label l = new Label( Messages.getString( "schedule.logLevel" ) );
+        l.setStyleName( "endLabel" ); //$NON-NLS-1$
+        logLvlHP.add( l );
+        dVP.add( logLvlHP );
+        detailPanel.add( dVP );
+
+        detailsLabel = new ErrorLabel( detailPanel );
+        VerticalFlexPanel detailVP = new VerticalFlexPanel();
+        detailVP.add( detailsLabel );
+        initWidget( detailVP );
+    }
+
+    private ListBox createLogLevelListBox() {
+        ListBox l = new ListBox();
+        for ( int ii = 0; ii < TimeUtil.LogLevel.length(); ++ii ) {
+            TimeUtil.LogLevel log = TimeUtil.LogLevel.get( ii );
+            l.addItem( Messages.getString( log.toString() ) );
+        }
+        return l;
+    }
+
+    @Override
+    public void setOnChangeHandler( ICallback<IChangeHandler> handler ) {
+        this.onChangeHandler = handler;
+    }
+
+    public boolean getEnableSafeMode() {
+        return enableSafeMode.getValue();
+    }
+
+    public void setEnableSafeMode( boolean enableSafeMode ) {
+        this.enableSafeMode.setValue( enableSafeMode );
+    }
+
+    public boolean getGatherMetrics() {
+        return gatherMetrics.getValue();
+    }
+
+    public void setGatherMetrics( boolean gatherMetrics ) {
+        this.gatherMetrics.setValue( gatherMetrics );
+    }
+
+    public String getLogLevelLb() {
+        return logLevelLb.getSelectedItemText();
+    }
+
+    public void setLogLevelLb( String logLevelLb ) {
+        for ( int ii = 0; ii < TimeUtil.LogLevel.length(); ++ii ) {
```

Comment:
Please, use either a single letter variable, such as the common `i`, or a more descriptive name such as `index`.